### PR TITLE
[MNT] Fix failing xgboost tests in the otherml test pipeline

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

The xgboost tests are currently failing because `use_label_encoder` is deprecated. From v1.7.0 onwards it should be omitted.

While I check the requirements I also found that we had several packages with minimum versions that are apparently not supported by Python 3.8, so I bumped them.

## Tests

- [x] no new tests required
- [ ] new tests added
- [x] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: test_othermlpackages/conda-xgboost.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ dependencies:
 - pip>=19.1.1
 - pytest
 - pytest-cov
-- xgboost
+- xgboost>=1.7.0
 - pandas<1.4 # Capped for xgboost only
```
Comment (hildeweerts) [inline test_othermlpackages/conda-xgboost.yaml:11]: Do you know if the pandas cap is still required if we set xgboost >=1.7.0?

**Review by hildeweerts — APPROVED** [review]

File: test_othermlpackages/conda-xgboost.yaml (hunk 1 above)
Comment (romanlutz) [reply test_othermlpackages/conda-xgboost.yaml:11]: Let's remove it and see what happens...

Comment (romanlutz) [inline test_othermlpackages/conda-xgboost.yaml:11]:
```suggestion
- pandas
```

Comment (romanlutz) [reply test_othermlpackages/conda-xgboost.yaml]: Still works, keeping the change 🙂

Comment (hildeweerts) [reply test_othermlpackages/conda-xgboost.yaml]: Awesome. Feel free to merge then if you agree with your own PR ;)

**Pull request merged**